src/Control/Optics/Linear/Internal.hs

```diff
@@ -132,30 +133,37 @@ build (Optical l) x = Linear.runCoKleisli (l (Linear.CoKleisli getConst')) (Cons
 getConst' :: Const a b #-> a
 getConst' (Const x) = x
 
-lengthOf :: MultIdentity r => Optic_ (NonLinear.Kleisli (Const (Adding r))) a b s t -> s -> r
+lengthOf :: MultIdentity r => Optic_ (NonLinear.Kleisli (Const (Adding r))) s t a b -> s -> r
 lengthOf l s = getAdded (gets l (const (Adding one)) s)
 
 -- XXX: the below two functions will be made redundant with multiplicity
 -- polymorphism on over and traverseOf'
-over' :: Optic_ (->) a b s t -> (a -> b) -> s -> t
+over' :: Optic_ (->) s t a b -> (a -> b) -> s -> t
 over' (Optical l) f = l f
 
-traverseOf' :: Optic_ (NonLinear.Kleisli f) a b s t -> (a -> f b) -> s -> f t
+traverseOf' :: Optic_ (NonLinear.Kleisli f) s t a b -> (a -> f b) -> s -> f t
 traverseOf' (Optical l) f = NonLinear.runKleisli (l (NonLinear.Kleisli f))
 
-iso :: (s #-> a) -> (b #-> t) -> Iso a b s t
+iso :: (s #-> a) -> (b #-> t) -> Iso s t a b
 iso f g = Optical (dimap f g)
 
-withIso :: Optic_ (Exchange a b) a b s t -> ((s #-> a) -> (b #-> t) -> r) -> r
+withIso :: Optic_ (Exchange a b) s t a b -> ((s #-> a) -> (b #-> t) -> r) -> r
 withIso (Optical l) f = f fro to
   where Exchange fro to = l (Exchange id id)
 
-withPrism :: Optic_ (Market a b) a b s t -> ((b #-> t) -> (s #-> Either t a) -> r) -> r
+withPrism :: Optic_ (Market a b) s t a b -> ((b #-> t) -> (s #-> Either t a) -> r) -> r
 withPrism (Optical l) f = f b m
   where Market b m = l (Market id Right)
 
-withLens :: Optic_ (Linear.Kleisli (Compose ((,) a) (FUN 'One b))) a b s t -> s #-> (a, b #-> t)
-withLens (Optical l) s = getCompose (Linear.runKleisli (l (Linear.Kleisli (\a -> Compose (a, id)))) s)
+-- XXX: probably a direct implementation would be better
+withLens
+  :: Optic_ (Linear.Kleisli (Compose ((,) a) (FUN 'One b))) s t a b
+  -> (forall c. (s #-> (c, a)) -> ((c, b) #-> t) -> r)
+  -> r
+withLens l k = k (Bifunctor.swap . (reifyLens l)) (uncurry ($))
```

Comment:
This is dense but ok.

Reply:
There is probably a better definition. One that doesn't indirect through `reifyLens`. I just went for the fastest implementation I could do (though I do find this particular definition illuminating, lenses are a deep and fascinating subject).